segmentstore/server/src/test/java/io/pravega/segmentstore/server/containers/MetadataStoreTestBase.java

```diff
@@ -718,10 +743,42 @@ public void testRegisterPinnedSegment() {
         Assert.assertTrue(context.connector.getContainerMetadata().getStreamSegmentMetadata(segmentId).isPinned());
     }
 
+    /**
+     * Verifies that a {@link SegmentType} marked as {@link SegmentType#isTransientSegment()} is rejected if it has an improperly
+     * formatted name.
+     *
+     * @throws ExecutionException ExecutionException if any occurs.
+     * @throws InterruptedException InterruptedException if any occurs.
+     * @throws TimeoutException TimeoutException if any occurs.
+     */
+    @Test
+    public void testTransientSegmentFormat() throws ExecutionException, InterruptedException, TimeoutException {
+        final String validTransientSegment = "scope/stream/transient#transient.00000000000000000000000000000000";
+        final String invalidTransientSegment = "scope/stream/transient";
+        @Cleanup
+        TestContext context = createTestContext();
+        // Should complete successfully.
+        context.getMetadataStore().createSegment(validTransientSegment, SegmentType.TRANSIENT_SEGMENT, new ArrayList<>(), TIMEOUT).get();
+        // Make sure our Transient Segment name maps to a valid Segment ID.
+        TestUtils.await(() -> {
+            return context.getMetadata().getStreamSegmentId(validTransientSegment, false) != ContainerMetadata.NO_STREAM_SEGMENT_ID;
+        }, 100, 1000);
```

Comment:
Give this thing a bit more timeout. In GitHubActions, this may take more than 1s due to other things running on the machine. 10-30 seconds timeout should be good and will ensure that your test doesn't fail sporadically.